src/components/header/Header.component.tsx

```diff
@@ -10,19 +10,26 @@ import styles from './Header.module.css';
 interface HeaderProps {
     onShowAbout: () => void;
     onShowProducts: () => void;
+    showTheme: () => void;
+    theme: 'light' | 'dark';
```

Comment:
Try to avoid "theme" property. You have "className={`${theme}-theme`}". Try to change styles inside css class.